docs/docs/services/qix-engine/access-control.md

```diff
@@ -54,64 +143,89 @@ Change owner | Change the owner of a resource.
-  "name": "John Doe",
-  "employeeType": "developer",
-  "tags": ["research"],
-  "custom": {
-    "country": "sweden"
-  }
-}
+If the resource type is `"App.Object"`, `resource` has the following additional attributes:
+
+Attribute | Description | Example condition
+--------- | ----------- | -----------------
+`resource.approved` | Indicator of whether the object was part of the original document when the document was published. Values: `true` or `false`. | `resource.approved="true"`
+`resource.description` | The object description. | `resource.description="My custom description"`
+`resource.objectType` | The object type. | `resource.objectType="field" or resource.objectType="my-generic-object"`
+`resource.published` | Indicator of whether the object is published. Values: `true` or `false`. | `resource.published="false"`
+`resource.app.name` | Name of the document that the object is part of. | `resource.app.name="Q3_Report"`
+
+### Resource Built-in Functions
+
+#### `HasPrivilege(<action>)`
+
+Boolean function for resource conditions that returns `true` if the user making the request has the specified access
+right for the targeted resource or resources. Otherwise returns `false`.
+
+##### Syntax
+
+```c
+resource.HasPrivilege(ACTION)
 ```
 
-Example usage in condition:
+The required parameter `ACTION` shall have a value of any of the supported actions described in [Actions](#Actions).
+
+##### Examples
+
+_None at the moment._
+
+#### `Empty()`
+
+Boolean function for resource conditions that returns `true` only if the specified resource has no connections
+(that is, has no value).
+
+##### Syntax
 
 ```c
-// Evaluates to `true` if the JWT looks like the example above:
-user.employeeType = "developer" and user.custom.country = "sweden"
+resource.resourcetype.Empty()
 ```
 
-### App
+##### Examples
 
-The document (called App in the rules syntax) context.
+_None at the moment._
 
-It has no additional attributes.
+#### `IsOwned()`
 
-### App.Object
+Boolean function for resource conditions that returns `true` only if the specified resource has an owner.
 
-An object inside a document context.
+##### Syntax
 
-Attribute | Description | Example condition
---------- | ----------- | -----------------
-approved | Indicator of whether the object was part of the original document when the document was published. Values: `true` or `false`. | `resource.approved="true"`
-description | The object description. | `resource.description="My custom description"`
-objectType | The object type. | `resource.objectType="field" or resource.objectType="my-generic-object"`
-published | Indicator of whether the object is published. Values: `true` or `false`. | `resource.published="false"`
-app.name | Name of the document that the object is part of. | `app.name="Q3_Report"`
+```c
+resource.IsOwned()
+```
+
+##### Examples
+
+```c
+// Evaluates to `true` if the resource is owned by the user
+// being evaluated:
+resource.IsOwned() and resource.owner = user
```

Comment:
Isn't the `IsOwned()` check unnecessary in this example since we also check the `owner` attribute?